app/controllers/teams_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TeamsController < ApplicationController
```

Comment:
Don't we need to add `before_action :require_login` to ensure that only logged in users may do this actions?

Reply:
It sure looks that way. I mean, or not... It's very common that these kind of websites allow some level of read-only  access to team data, etc. (maybe restricting some info).
What is indisputable is that we cannot send messages to `current_user` like we do on the view for joining/leaving without checking if the result of that accessor is `present?` (i.e. if some user is logged in).